monitoring/defaults_linux.go

```diff
@@ -116,8 +116,11 @@ func DefaultBootConfigParams() health.Checker {
 		BootConfigParam{
 			// https://cateee.net/lkddb/web-lkddb/NF_NAT_IPV4.html
 			// CONFIG_NF_NAT_IPV4 has been removed as of kernel 5.1
-			Name:             "CONFIG_NF_NAT_IPV4",
-			KernelConstraint: KernelVersionLessThan(KernelVersion{Release: 5, Major: 1}),
+			Name: "CONFIG_NF_NAT_IPV4",
+			KernelConstraints: []KernelConstraintFunc{
+				KernelVersionLessThan(KernelVersion{Release: 5, Major: 1}),
+				KernelVersionNotEquals(KernelVersion{Release: 4, Major: 18, Minor: 0, Patch: 240}),
```

Comment:
How would we go about implementing feature detection here? It doesn't looks like these configs are being used anywhere in Gravity, these configs are just required for running Kubernetes?

Reply:
> these configs are just required for running Kubernetes?

Correct.  Netfilter NAT is used the software defined networking functionality of Kubernetes -- needed for transparent pod-to-pod packet forwarding (best as I know).

For feature detection, I'd suggest we check the kernel modules.  If `nf_nat_ipv4` is present we then expect `CONFIG_NF_NAT_IPV4` to be set.  Compare:

Centos 8.2
```
Linux walt-rhel82-node-0 4.18.0-193.28.1.el8_2.x86_64 #1 SMP Thu Oct 22 00:20:22 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
[centos@walt-rhel82-node-0 ~]$ lsmod | grep nf_nat_ipv4
nf_nat_ipv4            16384  2 nft_chain_nat_ipv4,iptable_nat
nf_nat                 36864  2 nf_nat_ipv6,nf_nat_ipv4
nf_conntrack          155648  6 nf_conntrack_ipv6,nf_conntrack_ipv4,nf_nat,nft_ct,nf_nat_ipv6,nf_nat_ipv4
```

RHEL 8.3:
```
[redhat@walt-rhel-node-0 ~]$ uname -a
Linux walt-rhel-node-0 4.18.0-240.1.1.el8_3.x86_64 #1 SMP Fri Oct 16 13:36:46 EDT 2020 x86_64 x86_64 x86_64 GNU/Linux
[redhat@walt-rhel-node-0 ~]$ lsmod | grep nf_nat_ipv4
[redhat@walt-rhel-node-0 ~]$ lsmod | grep nf_nat
nf_nat                 45056  1 nft_chain_nat
nf_conntrack          172032  2 nf_nat,nft_ct
libcrc32c              16384  3 nf_conntrack,nf_nat,xfs
```

The nice thing about feature detection is if we get other oddball patched kernels, we won't have to write one-offs for them too.  The code will automatically determine the module isn't present & thus infer that these parameters don't need to be set.

Big caveat: I haven't looked at all at the satellite code & assessed how feasible it would be to do feature detection.  It may be quicker to put in some brittle rhel version code and run with that.  This isn't a super high return project, so we may want to adjust our investment accordingly.

I'd be glad to do some paring with you if you want to work on this together.

Reply:
Having not really thought about this much, a couple of other options to consider.

1. Use the existing mechanism that is based on kernel release, but detect redhat/centos kernels and after a specific version to ignore the field. So if kernel after 4.18.0 if distro is redhat then ignore the parameter.
2. Take the risk, and drop the check entirely, as I don't believe this is a config parameter we've seen missing multiple times. Might be once or twice a customer has built their own kernel missing this option.


Going the route of module detection, IIRC needs to also check if the module was done as builtin. I think our module detection code already does this, just an additional detail that needs to be considered.

Reply:
>     2. Take the risk, and drop the check entirely, as I don't believe this is a config parameter we've seen missing multiple times. Might be once or twice a customer has built their own kernel missing this option.

Rancher dropped the check entirely, FWIW: https://github.com/rancher/k3s/pull/2358/files

Reply:
@wadells yeah we can work on this together. I thought this was all we needed, but I guess we'll need to do a bit more work on this.